### ISSUE TITLE
State indicators

### ISSUE THREAD

**Issue by Fomentador** [BibliothecaDAO/eternum#72]
The happiness and defensive capability indicators for the Realms are displaying incorrect states:

![moodles-binary](https://github.com/BibliothecaDAO/eternum/assets/110478008/0ba4ba15-4801-476d-9386-039be35da840)

1 . Defence is not a binary state, and "vulnerable/not vulnerable" doesn't say much. Defensive armies can range from 1 to 30 battalions, so if you build 1 battalion you're still quite vulnerable, while if you build 20 you're neither vulnerable nor overly secure. Defence should be a % indicator ranging from 0 protection to 100% protection - I quite like the idea of a ring representation for the percentage, but maybe you think another design works better @Amaro-Koberle:

![Untitled](https://github.com/BibliothecaDAO/eternum/assets/110478008/749af81f-fcba-41e8-a214-6776653e6f3e)

The slightly different colours represent the percent of light battalions/heavy battalions, something that should be indicated by hover text: "18 Defensive Battalions 6 Heavy 12 Light". This indicator will be blurred in the present version, but we should have it ready either way.

2 . While happiness has indeed a base number and _it's_ a binary effect, its gameplay effects aren't. It matters less if you're happy or unhappy and more **how** happy or unhappy you're. Thus, this indicator would be best represented as a number below or above 0, ideally accompanied by colours and a descriptor state. For instance:

Below -30 "In despair" Red Gradient 4
-15 to -30 "Very unhappy" Red Gradient 3
-5 to -15 - "Unhappy" Red Gradient 2
0 to -5 - "Discontent" - Red Gradient 1
0 to 5 - "Content" - Green Gradient 1
5 to 15 - "Happy" - Green Gradient 2
15 to 30 - "Very Happy" - Green Gradient 3
30+ - "Exuberant" - Green Gradient Gradient 4

Numbers are placeholders. Colours are examples, @Amaro-Koberle will know more about colour combinations than I.

The happiness UI will be further changed in the future once I design the system for it, but these modifications will most likely remain.

**Comment by Amaro-Koberle** [comment]
Great feedback @Fomentador.
Rings are nice if you have a lot of space. This UI element is predominantly horizontal.

I suggest two bars, one for defense and one for happiness. The happiness bar has a mid point and can - [ ] represent negative values while the defense bar just goes from none to max.

These bars will have to be short to fit, otherwise they will make the whole container wider (realm detail panel). I think that's fine. We can move the status name (string) to a tooltip, or we could potentially give this element two rows (1 & 2) and two columns (A & B). Where 1A is a shield icon and the status name of the defense army; 2A is the smiley icon and the status name of the happiness level; 1B is the bar (none to max) representing defense army status; and 2B is the bar (min -> neutral -> max) representing the happiness level.

**Comment by Amaro-Koberle** [comment]
Colors red -> orange -> green. This is standard and easy to understand.

**Comment by Fomentador** [comment]
@Amaro-Koberle as I imagined you understood the use of space better than I, bars it's then. 

We don't really need a status name for defence since people understand a percentage of total number very easily. Saying "reasonably protected" is actually more subjective than just "50% protected", so adding descriptors there would end up making it more confusing. The tooltip there will be the composition of your defensive army - very relevant data that you don't have access to only by glancing at the icon. It should be a just a tad more descriptive than what I wrote, though, I'll edit the text to fix it.

Happiness is different because the values are completely arbitrary instead of a 0-100% scale, so players need help understanding what each number means. If you think we don't have enough space to add the strings then sure, we can dump them into the tooltip, but we'll probably need the tooltip to actually explain the system once it's implemented. What happiness does in the game, exactly? It's pretty evident what a defensive army does, but if I jump into a new game I have no idea which effects happiness has there. The tooltip exists so the player can know without having to leave the game and come ask in the discord. Right now happiness isn't implemented, so we can just cross that bridge when we get there, if you feel best. 

But by using descriptors only for happiness, don't you think we could use the same amount of space we currently use? Doing something like this:

![moodles_new_happiness](https://github.com/BibliothecaDAO/eternum/assets/110478008/a14ef5b3-f3db-42af-a8ed-803551284a60)

**Comment by Amaro-Koberle** [comment]
Yes, good points. We can ditch the defense status name and just use the space for a label reading "Defense".

And you are right that tooltips are crucial.
In fact, I think we should plan a session to really deck out all the tooltips in the game and add a lot of explanations for potentially confused players. Will think about it, that's another issue.

**Comment by Fomentador** [comment]
Lol, I'm writing an Issue about tooltips right now, it has a list there so we'll be able talk about it and add more to it. Still plenty of things I had written down for the polish phase that I'll be converting into git issues.

**Comment by Amaro-Koberle** [comment]
@blackmirrordesigner Check this out. Do you think you could do a figma mock up for @r0man1337 to implement these two bars?

**Comment by blackmirrordesigner** [comment]
two bars coming right up, in like 1-2 days please @Amaro-Koberle
